hypothesis-python/src/hypothesis/internal/intervalsets.py

```diff
@@ -69,3 +71,23 @@ def index_above(self, value):
             if value <= v:
                 return offset + (value - u)
         return self.size
+
+    def intersection(self, other):
```

Comment:
Rebase on master?  I recently added an `intersection` method which this should replace, mine is much slower 😅